Examples/UIComponentExample/Examples/Irregular Layout/Components/IrregularComponents.swift

```diff
@@ -0,0 +1,162 @@
+//
+//  IrregularComponents.swift
+//  UIComponentExample
+//
+//  Created by y H on 2021/7/29.
+//  Copyright © 2021 Luke Zhao. All rights reserved.
+//
+
+import UIKit
+import UIComponent
+
+struct CoverModel: Equatable {
+  let id = UUID().uuidString
+  let cover: URL = randomWebImage()
+}
+
+enum LayoutType: CaseIterable {
+  case topLeft, topRight, bottomLeft, bottomRight, layout1, layout2
+}
+
+struct IrregularSimpleComponent: ComponentBuilder {
+  let coverModels: [CoverModel]
+  let type: LayoutType
+  let spacing: CGFloat
+  let showOrder: Bool
+  
+  init(coverModels: [CoverModel], type: LayoutType, spacing: CGFloat, showOrder: Bool = false) {
+    self.coverModels = coverModels
+    self.type = type
+    self.spacing = spacing
+    self.showOrder = showOrder
+  }
+  
+  func build() -> Component {
+    CustomizeLayout {
+      for (index, item) in coverModels.enumerated() {
+        AsyncImage(item.cover).contentMode(.scaleAspectFill).clipsToBounds(true).id(item.id).overlay(showOrder ? Text("\(index)").textColor(.white).textAlignment(.center).backgroundColor(UIColor.black.withAlphaComponent(0.5)) : Space())
+      }
+    } blockFrames: { constraint in
+      return Array(genFrames(side: min(constraint.maxSize.width, constraint.maxSize.height), spacing: spacing, type: type).prefix(coverModels.count))
+    }
+  }
+}
+
+class IrregularCustomComponent: ComponentBuilder {
```

Comment:
This seems more like a `Layout` than a `Component`. It calculates all the frames by itself. Let's remove the `CustomLayout`, and make this the layout.
Let's also change the name to `HorizontalGallery` and `VerticalGallery`.

I would structure the usage of this to something like
```
struct GalleryItem: ComponentBuilder {
  let item: CoverItem
  let showOrder: Bool
  func build() -> Component {
    AsyncImage(item.cover)
      .contentMode(.scaleAspectFill)
      .clipsToBounds(true)
      .id(item.id)
    if showOrder {
      return image.overlay()
    } else {
      return image
    }
  }
}
struct GalleryIndexOverlay: ComponentBuilder {
  let index: Int
  func build() -> Component {
    Text("\(index)").textColor(.white).textAlignment(.center).backgroundColor(UIColor.black.withAlphaComponent(0.5))
  }
}
VerticalGallery {
  for (index, item) in coverModels.enumerated() {
    if showIndex {
      GalleryItem(item: item).overlay(GalleryIndexOverlay(index: index))
    } else {
      GalleryItem(item: item)
    }
  }
}
```